File: sistema-encuestas-frontend/src/app/services/respuesta.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Respuesta } from '../models/respuesta';

@Injectable({
  providedIn: 'root'
})
export class RespuestaService {
  private apiUrl = 'http://localhost:8090/api/respuestas';

  constructor(private http:HttpClient) { }

  agregarRespuestaAPregunta(preguntaId:number,contenido:string):Observable<Respuesta>{
    return this.http.post<Respuesta>(`${this.apiUrl}/agregar/${preguntaId}`,{id:preguntaId,contenido:contenido});
  }

  obtenerRespuestaPorPregunta(preguntaId:number):Observable<Respuesta[]> {
    return this.http.get<Respuesta[]>(`${this.apiUrl}/pregunta/${preguntaId}`);
  }

  obtenerDetallesRespuesta(preguntaId:number):Observable<Respuesta> {
    return this.http.get<Respuesta>(`${this.apiUrl}/${preguntaId}`);
  }

  actualizarRespuesta(respuestaId:number,nuevoContenido:string):Observable<Respuesta>{
    return this.http.put<Respuesta>(`${this.apiUrl}/${respuestaId}`,{id:respuestaId,contenido:nuevoContenido});
  }

  eliminarRespuesta(respuestaId:number):Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${respuestaId}`);

  }
}

File: Sistema-de-Encuestas/src/main/java/com/sisencuesta/services/RespuestaService.java
package com.sisencuesta.services;

import com.sisencuesta.models.Respuesta;

import java.util.List;
import java.util.Optional;

public interface RespuestaService {

    Respuesta agregarRespuestaAPregunta(Long preguntaId, String contenido);
    List<Respuesta>obtenerRespuestaPorPregunta(Long preguntaId);
    Optional<Respuesta>obtenerDetallesRespuesta(Long respuestaId);
    Respuesta actualizarRespuesta(Long respuestaId, String nuevoContenido);
    void eliminarRespuesta(Long respuestaId);
}
